AstronomyObjC/AstronomyObjC/Models/NSObject+NSJSONSerialization.m

```diff
@@ -0,0 +1,27 @@
+//
+//  NSObject+NSJSONSerialization.m
+//  AstronomyObjC
+//
+//  Created by Christopher Devito on 5/18/20.
+//  Copyright © 2020 Christopher Devito. All rights reserved.
+//
+
+#import "NSObject+NSJSONSerialization.h"
+#import "TMCMarsPhotoReference.h"
+#import "TMCMarsPhotoReference+NSJSONSerialization.h"
+
+@implementation NSObject (NSJSONSerialization)
+
+- (instancetype)initArrayWithDictionary:(NSDictionary *)dictionary {
+    self = [self init];
+    NSMutableArray *photoArray = [[NSMutableArray alloc] init];
+    NSArray *photos = dictionary[@"photos"];
+    for (int i = 0; i < photos.count; i++) {
+        TMCMarsPhotoReference *photo = [[TMCMarsPhotoReference alloc] initWithMarsPhotoDictionary:photos[i]];
+        [photoArray addObject:photo];
+    }
+    return photoArray;
+}
```

Comment:
Since you're already allocating the memory for `photoArray` on line 17, calling this method `init` might be confusing, since you do **not** want to call `alloc` when using this initializer (otherwise you'll be allocating twice as much memory as you need). Something like `photosArrayWithDictionary` might be more suitable.

Additionally, I'm not sure that `NSObject` is the best place to put this method; it might be best to put it as a method on a model controller class.